org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/DocumentLifeCycleHandler.java

```diff
@@ -383,6 +387,10 @@ public void handleClosed(DidCloseTextDocumentParams params) {
 			}
 			if (JDTUtils.isDefaultProject(unit) || !JDTUtils.isOnClassPath(unit) || unit.getResource().isDerived()) {
 				new DiagnosticsHandler(connection, unit).clearDiagnostics();
+			} else if (unit.hasUnsavedChanges()) {
+				unit.discardWorkingCopy();
+				unit.becomeWorkingCopy(new NullProgressMonitor());
```

Comment:
does it need to become a working copy if it's closed?

Reply:
Yep, I tried different API calls combination for this issue and this calling sequence meets our need.  And the reconcile have to be work on a  workingCopy.

Not familiar with CU/WorkingCopy, might have other more decent API available here?